shared/src/main/scala/amf/core/client/scala/adoption/IdAdopter.scala

```diff
@@ -72,20 +71,21 @@ class IdAdopter(
   }
 
   /** this is done specifically because of RAML scalar valued nodes, extension is only stored in annotation contained in
-   * AmfScalar and needs to have id defined due to potential validations
-   */
+    * AmfScalar and needs to have id defined due to potential validations
+    */
   private def traverseDomainExtensionAnnotation(scalar: AmfScalar, id: String): Seq[PendingAdoption] = {
     scalar.annotations.collect[PendingAdoption] { case domainAnnotation: DomainExtensionAnnotation =>
-      val extension = domainAnnotation.extension
+      val extension   = domainAnnotation.extension
       val generatedId = idMaker.makeId(id, extension.componentId)
       PendingAdoption(extension, generatedId)
     }
   }
 
-  private def getOrderedFields(obj: AmfObject): Iterable[FieldEntry] = {
+  // Need to override this filter from APB in order to avoid re-adopt modules from dependencies
+  protected def getOrderedFields(obj: AmfObject): Iterable[FieldEntry] = {
```

Comment:
Why not favour composition? In this way we don't have to create a new "IdAdopter" class, we just impleemnt a new ordering criteria